core/common/src/main/java/alluxio/conf/NestedAlluxioConfiguration.java

```diff
@@ -0,0 +1,178 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.conf;
+
+import com.google.common.collect.ImmutableMap;
+
+import java.time.Duration;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import javax.annotation.concurrent.ThreadSafe;
+
+/**
+ * Configuration that wraps around another configuration
+ * Priority for the value of a property follows:
+ * 1. outer configuration
+ * 2. inner configuration
+ */
+@ThreadSafe
+public class NestedAlluxioConfiguration implements AlluxioConfiguration {
+  /**
+   * Runtime level configuration.
+   */
+  private final AlluxioConfiguration mOuterConf;
+  /**
+   * Default configuration.
+   */
+  private final AlluxioConfiguration mInnerConf;
+
+  /**
+   * Constructs a new instance with the specified references without copying the underlying
+   * properties.
+   *
+   * @param outerConf the runtime level configuration to override
+   * @param innerConf the default configuration
+   */
+  public NestedAlluxioConfiguration(AlluxioConfiguration outerConf,
+      AlluxioConfiguration innerConf) {
+    mOuterConf = outerConf;
+    mInnerConf = innerConf;
+  }
+
+  private AlluxioConfiguration conf(PropertyKey key) {
+    return mOuterConf.isSet(key) ? mOuterConf : mInnerConf;
```

Comment:
Does this mean a default value in the outer layer gets to override a value explicitly set by the user in the inner layer?